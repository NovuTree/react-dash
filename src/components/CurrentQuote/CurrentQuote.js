import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';
import './CurrentQuote.scss';

const propTypes = {
  quote: PropTypes.shape({
    text: PropTypes.string, 
    source: PropTypes.string, 
    liked: PropTypes.bool
  }),
  onLikeQuote: PropTypes.func.isRequired 
};

const CurrentQuote = ({ quote, onLikeQuote }) => {
  const { text, source, liked } = quote;
  let heartClassName = 'empty heart';
  
  if (liked) {
    heartClassName = 'heart';
  }
  
  return (
    <div className="Current-Quote">
      <blockquote>
        <p>{text}</p>
        <div className="quote-source">
          {source} <Icon onClick={onLikeQuote} name={heartClassName} />
        </div>
      </blockquote> 
    </div>
  );
};

CurrentQuote.propTypes = propTypes;

export default CurrentQuote;

  

