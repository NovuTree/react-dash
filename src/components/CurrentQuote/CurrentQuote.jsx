import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  color: white;
  position: fixed;
  left: 50%;
  transform: translateX(-50%);
  bottom: 10px;
  width: 100%;
  max-width: 700px;
  text-align: center;
  font-weight: 300;
  `;
const Quote = styled.p`
  font-size: 1.25em;
  text-shadow: 0 1px 5px rgba(0,0,0,.1);
  `;
const Source = styled.p`
  font-size: 1em;
  color: white;
  text-shadow: 0 1px 5px rgba(0,0,0,.1);
  .heart {
    &:hover {
      cursor: pointer;
    }
  }
`;
const CurrentQuote = ({ quotes, toggleLike, setCurrentQuote }) => {
  const handleLike = (feedName, id) => {
    toggleLike(feedName, id);
    setCurrentQuote(id);
  }
  const { quoteFeeds, currentFeed: feedName, currentQuoteId } = quotes;
  
  // Load random quote
  const currentFeed = quoteFeeds.find(feed => {
    // currentFeed for Default feed if no quotes in currentFeed
    return feed.feedName === feedName && feed.quotes.length;
  }) || quoteFeeds[0];
  const { quotes: currentQuotes } = currentFeed;

  // Set quote to render currentQuote or a random quote initially
  const quote = currentQuotes.find(quote => quote.id === currentQuoteId)
  || currentQuotes[Math.floor(Math.random() * currentQuotes.length)];

  const { id, text, source, liked } = quote;

  let heart = 'empty heart icon';
  if (liked) heart = 'heart icon';

  return (
    <Container>
      <blockquote>
        <Quote>{text}</Quote>
        <Source>
          {source}
          {' '}<i onClick={() => handleLike(feedName, id)} className={heart} />
        </Source>
      </blockquote> 
    </Container>
  )
};

CurrentQuote.propTypes = {
  quotes: PropTypes.object.isRequired,
  toggleLike: PropTypes.func.isRequired 
};
CurrentQuote.defaultProps = {
  quotes: {},
  toggleLike: () => {}
}

export default CurrentQuote;
