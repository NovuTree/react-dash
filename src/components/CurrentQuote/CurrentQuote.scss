
@import '../../styles/variables.scss';

.Current-Quote {
  position: fixed;
  left: 50%;
  transform: translateX(-50%);
  bottom: 10px;
  width: 100%;
  max-width: 700px;
  text-align: center;
  font-size: 1em;
  font-weight: 300;
  .quote-source {
    color: $silver;
  }
}
