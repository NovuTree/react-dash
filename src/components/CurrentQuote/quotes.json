[
  ["Life isn’t about getting and having, it’s about giving and being.", "Kevin Kruse", false],
  ["Whatever the mind of man can conceive and believe, it can achieve.", "Napoleon Hill", false],
  ["Strive not to be a success, but rather to be of value.", "Albert Einstein", false],
  ["Two roads diverged in a wood, and I—I took the one less traveled by, And that has made all the difference.", "Robert Frost", false],
  ["I attribute my success to this: I never gave or took any excuse.", "Florence Nightingale", false],
  ["You miss 100% of the shots you don’t take.", "Wayne Gretzky", false],
  ["I’ve missed more than 9000 shots in my career. I’ve lost almost 300 games. 26 times I’ve been trusted to take the game winning shot and missed. I’ve failed over and over and over again in my life. And that is why I succeed.", "Michael Jordan", false],
  ["The most difficult thing is the decision to act, the rest is merely tenacity.", "Amelia Earhart", false],
  ["Every strike brings me closer to the next home run.", "Babe Ruth", false],
  ["Definiteness of purpose is the starting point of all achievement.", "W. Clement Stone", false],
  ["We must balance conspicuous consumption with conscious capitalism.", "Kevin Kruse", false],
  ["Life is what happens to you while you’re busy making other plans.", "John Lennon", false],
  ["We become what we think about.", "Earl Nightingale", false],
  ["14.Twenty years from now you will be more disappointed by the things that you didn’t do than by the ones you did do, so throw off the bowlines, sail away from safe harbor, catch the trade winds in your sails.  Explore, Dream, Discover.", "Mark Twain", false],
  ["15.Life is 10% what happens to me and 90% of how I react to it.", "Charles Swindoll", false],
  ["The most common way people give up their power is by thinking they don’t have any.", "Alice Walker", false],
  ["The mind is everything. What you think you become.", "Buddha", false],
  ["The best time to plant a tree was 20 years ago. The second best time is now.", "Chinese Proverb", false],
  ["An unexamined life is not worth living.", "Socrates", false],
  ["Eighty percent of success is showing up.", "Woody Allen", false],
  ["Your time is limited, so don’t waste it living someone else’s life.", "Steve Jobs", false],
  ["Winning isn’t everything, but wanting to win is.", "Vince Lombardi", false],
  ["I am not a product of my circumstances. I am a product of my decisions.", "Stephen Covey", false],
  ["Every child is an artist.  The problem is how to remain an artist once he grows up.", "Pablo Picasso", false],
  ["You can never cross the ocean until you have the courage to lose sight of the shore.", "Christopher Columbus", false],
  ["I’ve learned that people will forget what you said, people will forget what you did, but people will never forget how you made them feel.", "Maya Angelou", false],
  ["Either you run the day, or the day runs you.", "Jim Rohn", false],
  ["Whether you think you can or you think you can’t, you’re right.", "Henry Ford", false],
  ["The two most important days in your life are the day you are born and the day you find out why.", "Mark Twain", false],
  ["Whatever you can do, or dream you can, begin it.  Boldness has genius, power and magic in it.", "Johann Wolfgang von Goethe", false],
  ["The best revenge is massive success.", "Frank Sinatra", false],
  ["People often say that motivation doesn’t last. Well, neither does bathing.  That’s why we recommend it daily.", "Zig Ziglar", false],
  ["Life shrinks or expands in proportion to one’s courage.", "Anais Nin", false],
  ["If you hear a voice within you say “you cannot paint,” then by all means paint and that voice will be silenced.", "Vincent Van Gogh", false],
  ["There is only one way to avoid criticism: do nothing, say nothing, and be nothing.", "Aristotle", false],
  ["Ask and it will be given to you; search, and you will find; knock and the door will be opened for you.", "Jesus", false],
  ["The only person you are destined to become is the person you decide to be.", "Ralph Waldo Emerson", false],
  ["Go confidently in the direction of your dreams.  Live the life you have imagined.", "Henry David Thoreau", false],
  ["When I stand before God at the end of my life, I would hope that I would not have a single bit of talent left and could say, I used everything you gave me.", "Erma Bombeck", false],
  ["Few things can help an individual more than to place responsibility on him, and to let him know that you trust him.", "Booker T. Washington", false],
  ["Certain things catch your eye, but pursue only those that capture the heart.", " Ancient Indian Proverb", false],
  ["Believe you can and you’re halfway there.", "Theodore Roosevelt", false],
  ["Everything you’ve ever wanted is on the other side of fear.", "George Addair", false],
  ["We can easily forgive a child who is afraid of the dark; the real tragedy of life is when men are afraid of the light.", "Plato", false],
  ["Teach thy tongue to say, “I do not know,” and thous shalt progress.", "Maimonides", false],
  ["Start where you are. Use what you have.  Do what you can.", "Arthur Ashe", false],
  ["When I was 5 years old, my mother always told me that happiness was the key to life.  When I went to school, they asked me what I wanted to be when I grew up.  I wrote down ‘happy’.  They told me I didn’t understand the assignment, and I told them they didn’t understand life.", "John Lennon", false],
  ["Fall seven times and stand up eight.", "Japanese Proverb", false],
  ["When one door of happiness closes, another opens, but often we look so long at the closed door that we do not see the one that has been opened for us.", "Helen Keller", false],
  ["Everything has beauty, but not everyone can see.", "Confucius", false],
  ["How wonderful it is that nobody need wait a single moment before starting to improve the world.", "Anne Frank", false],
  ["When I let go of what I am, I become what I might be.", "Lao Tzu", false],
  ["Life is not measured by the number of breaths we take, but by the moments that take our breath away.", "Maya Angelou", false],
  ["Happiness is not something readymade.  It comes from your own actions.", "Dalai Lama", false],
  ["If you’re offered a seat on a rocket ship, don’t ask what seat! Just get on.", "Sheryl Sandberg", false],
  ["First, have a definite, clear practical ideal; a goal, an objective. Second, have the necessary means to achieve your ends; wisdom, money, materials, and methods. Third, adjust all your means to that end.", "Aristotle", false],
  ["If the wind will not serve, take to the oars.", "Latin Proverb", false],
  ["You can’t fall if you don’t climb.  But there’s no joy in living your whole life on the ground.", "Unknown", false],
  ["We must believe that we are gifted for something, and that this thing, at whatever cost, must be attained.", "Marie Curie", false],
  ["Too many of us are not living our dreams because we are living our fears.", "Les Brown", false],
  ["Challenges are what make life interesting and overcoming them is what makes life meaningful.", "Joshua J. Marine", false],
  ["If you want to lift yourself up, lift up someone else.", "Booker T. Washington", false],
  ["I have been impressed with the urgency of doing. Knowing is not enough; we must apply. Being willing is not enough; we must do.", "Leonardo da Vinci", false],
  ["Limitations live only in our minds.  But if we use our imaginations, our possibilities become limitless.", "Jamie Paolinetti", false],
  ["You take your life in your own hands, and what happens? A terrible thing, no one to blame.", "Erica Jong", false],
  ["What’s money? A man is a success if he gets up in the morning and goes to bed at night and in between does what he wants to do.", "Bob Dylan", false],
  ["I didn’t fail the test. I just found 100 ways to do it wrong.", "Benjamin Franklin", false],
  ["In order to succeed, your desire for success should be greater than your fear of failure.", "Bill Cosby", false],
  ["A person who never made a mistake never tried anything new.", " Albert Einstein", false],
  ["The person who says it cannot be done should not interrupt the person who is doing it.", "Chinese Proverb", false],
  ["There are no traffic jams along the extra mile.", "Roger Staubach", false],
  ["It is never too late to be what you might have been.", "George Eliot", false],
  ["You become what you believe.", "Oprah Winfrey", false],
  ["I would rather die of passion than of boredom.", "Vincent van Gogh", false],
  ["A truly rich man is one whose children run into his arms when his hands are empty.", "Unknown", false],
  ["It is not what you do for your children, but what you have taught them to do for themselves, that will make them successful human beings.", "Ann Landers", false],
  ["If you want your children to turn out well, spend twice as much time with them, and half as much money.", "Abigail Van Buren", false],
  ["Build your own dreams, or someone else will hire you to build theirs.", "Farrah Gray", false],
  ["The battles that count aren’t the ones for gold medals. The struggles within yourself–the invisible battles inside all of us–that’s where it’s at.", "Jesse Owens", false],
  ["Education costs money.  But then so does ignorance.", "Sir Claus Moser", false],
  ["I have learned over the years that when one’s mind is made up, this diminishes fear.", "Rosa Parks", false],
  ["It does not matter how slowly you go as long as you do not stop.", "Confucius", false],
  ["If you look at what you have in life, you’ll always have more. If you look at what you don’t have in life, you’ll never have enough.", "Oprah Winfrey", false],
  ["Remember that not getting what you want is sometimes a wonderful stroke of luck.", "Dalai Lama", false],
  ["You can’t use up creativity.  The more you use, the more you have.", "Maya Angelou", false],
  ["Dream big and dare to fail.", "Norman Vaughan", false],
  ["Our lives begin to end the day we become silent about things that matter.", "Martin Luther King Jr.", false],
  ["Do what you can, where you are, with what you have.", "Teddy Roosevelt", false],
  ["If you do what you’ve always done, you’ll get what you’ve always gotten.", "Tony Robbins", false],
  ["Dreaming, after all, is a form of planning.", "Gloria Steinem", false],
  ["It’s your place in the world; it’s your life. Go on and do all you can with it, and make it the life you want to live.", "Mae Jemison", false],
  ["You may be disappointed if you fail, but you are doomed if you don’t try.", "Beverly Sills", false],
  ["Remember no one can make you feel inferior without your consent.", "Eleanor Roosevelt", false],
  ["Life is what we make it, always has been, always will be.", "Grandma Moses", false],
  ["The question isn’t who is going to let me; it’s who is going to stop me.", "Ayn Rand", false],
  ["When everything seems to be going against you, remember that the airplane takes off against the wind, not with it.", "Henry Ford", false],
  ["It’s not the years in your life that count. It’s the life in your years.", "Abraham Lincoln", false],
  ["Change your thoughts and you change your world.", "Norman Vincent Peale", false],
  ["Either write something worth reading or do something worth writing.", "Benjamin Franklin", false],
  ["Nothing is impossible, the word itself says, “I’m possible!”", "–Audrey Hepburn", false],
  ["The only way to do great work is to love what you do.", "Steve Jobs", false],
  ["If you can dream it, you can achieve it.", "Zig Ziglar", false]
]