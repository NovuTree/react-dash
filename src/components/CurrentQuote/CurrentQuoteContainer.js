// TODO: save quote to localStorage
import React, { Component } from 'react';
import { loadFromStorage, saveToStorage } from '../../localStorage';
import CurrentQuote from './CurrentQuote';

class CurrentQuoteContainer extends Component {
  constructor() {
    super();
    this.state = {
      quote: ''
    };
  }
  componentDidMount() {
    this.setCurrentQuote();
  }
  
  setCurrentQuote() {
    const defaultList = loadFromStorage('defaultList');
    if (defaultList) { // FIXME: deal with no defaultList
      const { quotes }  = defaultList;
      this.setState({
        quote: quotes[Math.floor(Math.random() * quotes.length)]
      });
    }
  }
  likeQuote = () => {
    const { quote } = this.state;

    this.setState({
      quote: Object.assign({}, quote, {
        liked: quote.liked ? false : true
      })
    });
    
  }
  render() {
    const { quote } = this.state;
    return (
      <CurrentQuote quote={quote} onLikeQuote={this.likeQuote} />
    );
  }
}

export default CurrentQuoteContainer; 
