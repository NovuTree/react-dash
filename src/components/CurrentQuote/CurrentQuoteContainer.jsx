import { connect } from 'react-redux';
import CurrentQuote from './CurrentQuote';
import { toggleLike } from '../../actions/QuotesActionCreators.js';

const mapStateToProps = state => {
  return {
    quotes: state.quotes
  };
};

const mapDispatchToProps = dispatch => {
  return {
    toggleLike: (feedName, id) => {
      dispatch(toggleLike(feedName, id));
    }
  };
};

const CurrentQuoteContainer = connect(
  mapStateToProps, 
  mapDispatchToProps
)(CurrentQuote);

export default CurrentQuoteContainer;
