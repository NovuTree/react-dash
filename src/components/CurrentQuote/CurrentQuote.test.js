import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CurrentQuote from './CurrentQuote';

it('should render CurrentQuote', () => {
  const tree = renderer.create(<CurrentQuote />).toJSON();
  expect(tree).toMatchSnapshot();
});
