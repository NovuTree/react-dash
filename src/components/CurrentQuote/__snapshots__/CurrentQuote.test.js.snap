// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentQuote 1`] = `
<div
  className="CurrentQuote"
>
  <blockquote>
    <p />
    <div
      className="quote-source"
    >
       
      <i
        aria-hidden="true"
        className="empty heart icon"
        onClick={undefined}
      />
    </div>
  </blockquote>
</div>
`;
